controller/identity/domain.go

```diff
@@ -0,0 +1,41 @@
+package identity
+
+import (
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+// TrustDomain is a namespace for identities.
+type TrustDomain struct {
+	controlNS, domain string
+}
+
+// NewTrustDomain creates a new identity namespace.
+func NewTrustDomain(controlNS, domain string) (*TrustDomain, error) {
+	if errs := validation.IsDNS1123Label(controlNS); len(errs) > 0 {
+		for _, e := range errs {
+			return nil, fmt.Errorf("Invalid label '%s': %s", controlNS, e)
+		}
+	}
+	if domain == "" {
```

Comment:
This check isn't needed as the regex inside `IsDNS1123Label()` doesn't allow empty strings.

Reply:
But the domain isn't restricted to be just a label... it needs to be a domain suffix.

Reply:
Changed this to use a kubernetes validator